recipes/libftdi/1.x/conanfile.py

```diff
@@ -0,0 +1,73 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class LibFtdiConan(ConanFile):
+    name = "libftdi"
+    description = "A library to talk to FTDI chips"
+    license = ["GNU LGPL v2.1"]
+    topics = ("conan", "libftdi1")
+    homepage = "https://www.intra2net.com/en/developer/libftdi/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package", "pkg_config"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = (
+            "libusb/1.0.23",
+            "boost/1.74.0"
+            )
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "libftdi1-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_cmakelists(self, subfolder):
+        cmakelists_path = os.path.join(self._source_subfolder, subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists_path, "CMAKE_SOURCE_DIR", "PROJECT_SOURCE_DIR", strict=False)
+        tools.replace_in_file(cmakelists_path, "CMAKE_BINARY_DIR", "PROJECT_BINARY_DIR", strict=False)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        options = {
+            "BUILD_TESTS": False,
+            "EXAMPLES": False,
+            "FTDI_EEPROM": False,
+            "FTDIPP" : True,
+            "STATICLIBS": not self.options.shared
+        }
+        self._cmake.definitions.update(options)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        self._patch_cmakelists("")
+        self._patch_cmakelists("ftdipp")
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        os.unlink(os.path.join(self.package_folder, "bin", "libftdi1-config"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "LibFTDI1"
+        self.cpp_info.names["cmake_find_package_multi"] = "LibFTDI1"
+        self.cpp_info.names["pkgconfig"] = "libftdi1"
+        self.cpp_info.includedirs.append(
+            os.path.join(self.package_folder, "include", "libftdi1")
+        )
+        self.cpp_info.libs = ["ftdipp1", "ftdi1"]
```

Comment:
You combine both libraries.
Does the cmake config file, installed by libftdi's cmake config (and removed by us), create separate targets for the c and c++ library?
Maybe this requires conan components?

Reply:
These batch of errors (the .pc related errors) show that you need to use components, 
and set the `pkg_config` names.

Reply:
I'm not sure it's possible to exactly mimic LibFTDI's CMake configuration with conan (Libftdi uses `LibFTDI1` as the package name but `LIBFTDI` and `LIBFTDIPP` as the package variable names), but I've attempted to get as close as I can with conan components using the targets `LibFTDI1::ftdi` and `LibFTDI1::ftdipp`.

Reply:
Conan has 3 settings

Filenames (sets the cmake package name, is find_package)
Names (the cmake namespace of the targets)
Components names (each unique target)

I am sure we have an example I just can't recall one.

Reply:
The `find_package` name is `LibFTDI1`, as used in `examples/cmake_example`. The built-in find package then generates the variables `$LIBFTDI_LIBRARIES` and `$LIBFTDIPP_LIBRARIES`,

However the variables generated by the conan-generated find package when using components are `${LibFTDI1_component_LIBRARIES}` etc. I'm not sure it's possible to create multiple top-level variable names with conan, with a different name from the `find_package` (I did try various combinations as you suggested).

The targets created by LibFTDI are not namespaced, which I know conan doesn't support. See also conan-io/conan#7615

Reply:
Yes since the variables fashion is outdated its unlikely that conan will add support for that. So yes it seems this can not be done.